### ISSUE TITLE
Stream returns no content (not yet implemented)

### ISSUE THREAD

**Issue by ggorlen** [chihebnabil/openai-api-mock#3]
### Expected behavior

The stream returns content when `stream: true` is added to the request payload with the mock enabled.

### Actual behavior

The stream does not return content when `stream: true` is added to the request payload with the mock enabled.

### Minimal reprex

package.json:

```json
{
  "dependencies": {
    "dotenv": "^16.4.4",
    "openai": "^4.48.2",
    "openai-api-mock": "^0.1.13"
  }
}
```

.env:

```
OPENAI_API_KEY=sk-0... (actual OpenAI API key)
```

index.js:

```js
const dotenv = require("dotenv");
const OpenAI = require("openai");
const { mockOpenAIResponse } = require("openai-api-mock");

// mockOpenAIResponse(true);
dotenv.config();
const openai = new OpenAI();

(async () => {
  const stream = true;
  const completion = await openai.chat.completions.create({
    messages: [{ role: "user", content: "hello" }],
    model: "gpt-4-turbo-preview",
    stream,
  });

  if (stream) {
    for await (const chunk of completion) {
      process.stdout.write(chunk.choices[0].delta.content ?? "")
    }
    console.log();
  } else {
    console.log(completion.choices[0].message.content);
  }
})();
```

Command: `npm i && node index` (tested on Ubuntu 22.04 with Node 20.11.1 and npm 10.2.4).

Reproduction steps:
1. Run the script without modification and observe the OpenAI response streamed as expected.
2. Run the script again after changing `stream = false` and observe the OpenAI response logged as a single JSON response, non-stream, as expected.
3. Now uncomment the mock (`// mockOpenAIResponse(true);`) and run it again with `stream = false`. Lorem ipsum text is printed as expected, ensuring the mock is working.
4. Finally, change `stream = true` with the mock still active. Observe that only a newline is printed and nothing is printed in the `for` loop that consumes the stream. This is unexpected--I was expecting a series of chunks of lorem ipsum text to be streamed to the console with a bit of delay between each chunk.

### Suggested solution

It appears that streaming hasn't been implemented yet. Maybe this can help motivate the use case. In the meantime, consider mentioning that streaming isn't supported in the readme.

Here's a very quick workaround mock that satisfies the above example code and might help kick off the feature:

```js
// mock-openai.js
const { setTimeout } = require("node:timers/promises");

const response = `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.

Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.`;

const randInt = (n) => ~~(Math.random() * n);

const makeChunk = (content) => ({
  choices: [{ delta: { content } }],
});

module.exports = function () {
  return {
    chat: {
      completions: {
        create({ stream }) {
          if (!stream) {
            return (async () => {
              await setTimeout(1000 + randInt(500));
              return { choices: [{ message: { content: response } }] };
            })();
          }

          return (async function* () {
            await setTimeout(1000);
            let lastI = 0;

            for (let i = 0; i < response.length; i += randInt(6) + 4) {
              await setTimeout(randInt(20) + 50);
              yield makeChunk(response.slice(lastI, i));
              lastI = i;
            }

            yield makeChunk(response.slice(lastI));
          })();
        },
      },
    },
  };
};
```

Use in the main file:

```js
const OpenAI =
  process.env.NODE_ENV === "development"
    ? require("./mock-openai")
    : require("openai");
```

**Comment by chihebnabil** [comment]
Thank you for reporting this issue! I want let you know that the mocking stream feature will be implemented this week. 
I appreciate your patience and support.

Best regards,
Nabil

**Comment by chihebnabil** [comment]
[ggorlen](https://github.com/ggorlen) i just published [0.0.14](https://www.npmjs.com/package/openai-api-mock?activeTab=readme) which support streaming

**Comment by ggorlen** [comment]
Wow, that was fast! Thanks!

Unfortunately, I think the stream only works once, then crashes. If you'd opened a PR, I'd have been happy to review it. I've opened a new ticket https://github.com/chihebnabil/openai-api-mock/issues/4 to handle the new issue.
